natlas-server/app/api/prepare_work.py

```diff
@@ -0,0 +1,38 @@
+from flask import current_app
+from ipaddress import ip_network
+import random
+import string
+from app.models import ScopeItem
+
+
+def get_target_tags(target):
+	targetnet = ip_network(target)
+	tags = []
+	for scope in current_app.ScopeManager.getScope():
+		if scope.overlaps(targetnet):
+			scopetags = ScopeItem.query.filter_by(target=str(scope)).first().tags.all()
+			for tag in scopetags:
+				tags.append(tag.name)
+	return list(set(tags)) # make it a set for only uniques, then make it a list to serialize to JSON
+
+
+def get_unique_scan_id():
+	scan_id = ''
+	while scan_id == '':
+		rand = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(16))
+		count, context = current_app.elastic.gethost_scan_id(rand)
+		if count == 0:
+			scan_id = rand
+	return scan_id
+
+
+def prepare_work(work):
+	work['tags'] = get_target_tags(work['target'])
+	work['type'] = 'nmap'
+	work['agent_config'] = current_app.agentConfig
+	work['agent_config']['scripts'] = current_app.agentScriptStr
+	work["services_hash"] = current_app.current_services["sha256"]
+	work['scan_id'] = get_unique_scan_id()
+	work['status'] = 200
+	work['message'] = "Target: " + str(work['target'])
```

Comment:
This message is a little weird. Why are passing back free text blobs back to the client?

Reply:
This message isn't really necessary anymore, originally the agent networking code would print out any message it got from the server because I wrote it super poorly and didn't define the different types of errors that could occur. I added the status and message to a successful work request in an effort to make the json blobs more consistent about what fields were in them. But I think we can move away from this in the next iterations as we improve the networking code and properly define the interface.